### DISCUSSION TITLE
How to get the fragmentation information?

### DISCUSSION THREAD

**Discussion by yangyangbian** [vdemichev/DiaNN#406] [Q&A]
Hi guys.
I used DIN-NN 1.8 to search my DIA raw files.
In the output files, I can not find files containing some information like "number of fragment ion per peptides".
Could you kindly show me how to get this information?
Do I need to re-run the DIA search with some new options?
![image](https://user-images.githubusercontent.com/107282849/173160247-a7f1c25d-ad1c-43c8-89f9-d96ea47ca998.png)

Thanks a lot.

**Comment by vdemichev** [comment]
This information is not necessary for a typical analysis. However you can get fragment info by looking at fragments annotated in 1ssDDA_0928_MQ-lib.tsv.

**Reply by yangyangbian** [reply]
Thanks a lot for the quick reply.
I found a column named as "Fragment.Quant.Raw" in this file.
There are multiple values in each cell.
Does one separate value equals to one fragment ion?
For example: 5.47769e+06; 2.26476e+06; 676365; 1.78135e+06; 481900; 72416.8; 0; 743767; 5.47769e+06; 3.40344e+06; 3.40833e+06; 3.08007e+06; 3.21578e+06; 2.21188e+06; 676365; 1.78135e+06; 292489; 0; 0; 743767;

I would like to ask more questions.
1) I re-call some softwares only keep top 6 (maximum) fragment ions for each peptides.
How many fragment ions does DIA-NN use to build the spectra library for each peptides?
2）If DIA-NN also uses top 6 fragment ions for DIA identification and quantificaiton.
Why there are more than 6 values in this cell?
3) I would like to analyze the distribution of the numbers of fragments matched to each peptide.
But, currently I can not find a solution.

Thanks a lot.
I am looking forward to your reply.

**Reply by vdemichev** [reply]
Yes, each value corresponds to fragment ion. Can get annotation of those fragments with --report-lib-info. 
1. 12 for in silico prediction from FASTA, unlimited for building a library from DIA data - depends on the quality of fragment signals. DIA-NN uses more than 6 fragments for identification and quant.
2. Can use the report-lib.tsv file.

Best,
Vadim
